File: src/main/java/com/yogesh/meditrack/service/AppointmentService.java
package com.yogesh.meditrack.service;

import com.yogesh.meditrack.entity.Appointment;

import java.util.List;

public interface AppointmentService {

    Appointment bookAppointment(Appointment appointment);

    List<Appointment> getAllAppointments();

    Appointment getAppointmentById(Long id);

    Appointment updateAppointment(Long id,
                                  Appointment appointment);

    void cancelAppointment(Long id);
}

File: frontend/src/service/appointmentService.js
import api from "./api";

export const getAllAppointments = () => api.get("/appointments");

export const getAppointmentById = (id) => api.get(`/appointments/${id}`);

export const bookAppointment = (appointment) =>
  api.post("/appointments", appointment);

export const updateAppointment = (id, appointment) =>
  api.put(`/appointments/${id}`, appointment);

export const cancelAppointment = (id) => api.delete(`/appointments/${id}`);